### DISCUSSION TITLE
IPOPT slow iterations due to “Reallocating memory for MA57: lfact”

### DISCUSSION THREAD

**Discussion by 0xF0F0F0F** [coin-or/Ipopt#540] [Q&A] [answered]
Hi, I am solving a large scale quadratic problem with quadratic equality constaint problem (QCQP):
min_x  x' * H * x
s.t. x'*Q*x = 1
       A x <= b
where H is a positive definite matrix, Q is a positive semi-definite matrix, and n is around 5000.

I tried to use IPOPT (ver 3.12.9) to solve the QCQP problem by providing all jacobian & hessian informations and a feasible initial guess x0. IPOPT could solve my problem very fast in most of the time. However, sometimes IPOPT iterations stucks for more than 10 mins! Because it keeps

"Reallocating memory for MA57: lfact"

I am wondering, is there any way to fix this problem?

**Comment by svigerske** [comment, marked as answer]
Hmm, I don't know. You could experiment with some options for MA57: https://coin-or.github.io/Ipopt/OPTIONS.html#OPT_MA57_Linear_Solver

**Reply by 0xF0F0F0F** [reply]
Thank you for your replying, I am wondering, is there any way to automatically terminate IPOPT interation once it stucks at 
"Reallocating memory for MA57: lfact"?

**Reply by svigerske** [reply]
No, not that I know.

**Reply by 0xF0F0F0F** [reply]
can option "max_wall_time" do the termination if iteration stucks too long?

**Reply by svigerske** [reply]
No, the timelimit is not checked within MA57 or in the loop where Ipopt figures out how much memory it needs to allocate for MA57.
The timelimit is only checked once per iteration.

**Reply by 0xF0F0F0F** [reply]
I solved my problem by setting "options.ipopt.ma57_automatic_scaling = 'yes';"

**Comment by 0xF0F0F0F** [comment]
thanks！

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
